python/grass/pygrass/modules/grid/grid.py

```diff
@@ -415,6 +422,9 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_value, exc_traceback):
         if exc_type:
             self._clean_func()
+            return
+        if self._clean:
```

Comment:
if `run()` method is called with param  arg `clean=False`

https://github.com/OSGeo/grass/blob/6b61d1087ba9cf7429df8830c636a827bc5c3f5d/python/grass/pygrass/modules/grid/grid.py#L640